http/validators.go

```diff
@@ -119,14 +119,16 @@ func (s *Service) Validators(ctx context.Context,
 		return nil, errors.Join(errors.New("cannot specify both indices and public keys"), client.ErrInvalidOptions)
 	}
 
-	if len(opts.Indices) == 0 && len(opts.PubKeys) == 0 {
-		// Request is for all validators; fetch from state.
-		return s.validatorsFromState(ctx, opts)
-	}
+	if !s.reducedMemoryUsage {
+		if len(opts.Indices) == 0 && len(opts.PubKeys) == 0 {
```

Comment:
This part (request is for all validators) should be outside of the condition check; it is more efficient to obtain validators from state if all are requested, as we can use SSZ to pull the state so this gives us a smaller memory footprint than fetching just the validators with JSON.